cli/src/opts/wallet/mod.rs

```diff
@@ -520,6 +520,25 @@ impl Signer for &WalletSigner {
     }
 }
 
+impl WalletSigner {
+    pub async fn sign_hash(&self, hash: H256) -> Result<Signature, WalletSignerError> {
+        match self {
+            Self::Local(inner) => inner.sign_hash(hash).map_err(Into::into),
+            // TODO: verify if Ledger & Trezor actually hashes the messages
+            Self::Ledger(inner) => inner.sign_message(hash).await.map_err(Into::into),
+            Self::Trezor(inner) => inner.sign_message(hash).await.map_err(Into::into),
+            Self::Aws(_) => todo!()
+            // Self::Aws(inner) => {
+            //     let sig = self.sign_digest(hash.into()).await.map_err(Into::into)?;
+            //     let mut sig =
+            //         sig_from_digest_bytes_trial_recovery(&sig, hash.into(), &self.pubkey);
+            //     sig.v = (self.chain_id() * 2 + 35) + sig.v;
+            //     Ok(sig)
+            // }
```

Comment:
I'm having some difficulties implementing `sign_hash` for Ledger, Trezor, and Aws wallets because it seems the hash signing methods are not exposed directly. Also, I'm not too sure how I can test these different methods locally, is there somewhere I can refer to in the repo?

For Ledger, I think it might be possible to use `sign_payload(INS::SIGN, payload)`, but a small issue here is `INS` isn't exposed.

For Trezor, using `Trezor::ethereum_sign_message` might be possible, but not too sure.

For Aws, using `sign_digest` should work.